### ISSUE TITLE
IO.defer(null) probably doesn't do what you expect

### ISSUE THREAD

**Issue by djspiewak** [typelevel/cats-effect#1178]
I haven't tried it, but I suspect that it hangs forever. It *should* be a `NullPointerException`. This is a scary surfacing of `IOFiber`'s internals.

**Comment by SystemFw** [comment]
>  but I suspect that it hangs forever. 

it does

**Comment by vasilmkd** [comment]
I can see that CE2 doesn't allow any `null`s to enter IO, whether it's through `delay` or `pure`. Should CE3 have the same semantics?

**Comment by djspiewak** [comment]
> I can see that CE2 doesn't allow any nulls to enter IO, whether it's through delay or pure. Should CE3 have the same semantics?

I believe it *does* allow nulls:

```
@ import $ivy.`org.typelevel::cats-effect:2.2.0` 
import $ivy.$                                 

@ cats.effect.IO.pure(null).unsafeRunSync() 
res1: Null = null

@ cats.effect.IO(null).unsafeRunSync() 
res2: Null = null
```

**Comment by vasilmkd** [comment]
I tried it in scastie, and it seemed to me that it didn't. My bad.

**Comment by vasilmkd** [comment]
Resolved by #1194.
